apps/mobile/src/services/key-storage/key-storage.service.ts

```diff
@@ -127,4 +140,35 @@ export class KeyStorageService implements IKeyStorageService {
 
     return DeviceCrypto.decrypt(keyName, encryptedPassword, iv, this.BIOMETRIC_PROMPTS.STANDARD)
   }
+
+  private async removeKey(userId: string, requireAuth: boolean): Promise<void> {
+    const keyName = this.getKeyNameDeviceCrypto(userId)
+    const service = this.getKeyService(userId)
+
+    // First, try to delete from keychain (requires authentication if enabled)
+    const keychainOptions: Keychain.GetOptions = { service }
+    if (requireAuth) {
+      keychainOptions.accessControl = Keychain.ACCESS_CONTROL.BIOMETRY_CURRENT_SET_OR_DEVICE_PASSCODE
+    }
+
+    try {
+      // Check if the key exists in keychain
+      const result = await Keychain.getGenericPassword(keychainOptions)
+      if (result) {
+        // Delete from keychain
+        await Keychain.resetGenericPassword({ service })
+      }
+    } catch (error) {
+      // If key doesn't exist, that's fine - we still want to try to remove from device crypto
+      Logger.warn('Key not found in keychain or authentication failed:', error)
+    }
+
+    // Try to remove the encryption key from device crypto
+    try {
+      await DeviceCrypto.deleteKey(keyName)
```

Comment:
Where is the difference between this and `Keychain.resetGenericPassword`? Is it both trying to delete the same key?